model-optimizer/extensions/middle/ssd_anchors_middle_replacer.py

```diff
@@ -0,0 +1,203 @@
+                ('reshape2', dict(op='Reshape')),
+                ('reshape2_data', dict(kind='data')),
+                ('reshape3', dict(op='Reshape')),
+                ('reshape3_data', dict(kind='data')),
+                ('concat', dict(op='Concat')),
+            ],
+            edges=[
+                ('slice_like', 'slice_like_data'),
+                ('slice_like_data', 'reshape0'),
+                ('reshape0', 'reshape0_data'),
+                ('reshape0_data', 'reshape1'),
+                ('reshape1', 'reshape1_data'),
+                ('reshape1_data', 'reshape2'),
+                ('reshape2', 'reshape2_data'),
+                ('reshape2_data', 'reshape3'),
+                ('reshape3', 'reshape3_data'),
+                ('reshape3_data', 'concat'),
+            ])
+
+    def replace_pattern(self, graph: Graph, match: Dict[str, Node]):
+
+        concat_node = match['concat']
+        concat_node['axis'] = 1
+
+        slice_like = match['slice_like']
+        slice_like_name = slice_like.soft_get('name', slice_like.id)
+        slice_like.out_port(0).disconnect()
+        match['reshape3'].out_port(0).get_connection().set_source(slice_like.out_port(0))
+        slice_like['override_output_shape'] = True
+
+        reshape = create_op_node_with_second_input(graph, Reshape, int64_array([1, -1]), op_attrs=dict(
+            name=slice_like_name + '/Reshape', override_output_shape=True))
+        slice_like.out_port(0).get_connection().insert_node(reshape)
+
+
+class SsdAnchorsMiddleReplacer(MiddleReplacementPattern):
+
+    """
+    Replacing sub-graph with all anchors to sub-graph which calculates prior boxes values by formulas:
+
+    value[i] = xmin = value[i] - (value[i + 2] / 2)
+    value[i + 1] = ymin = value[i + 1] - (value[i + 3] / 2)
+    value[i + 2] = xmax = value[i] + (value[i + 2] / 2)
+    value[i + 3] = ymax = value[i + 1] + (value[i + 3] / 2)
+    """
+
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['fw'] == 'mxnet' and graph.graph['cmd_params'].enable_ssd_gluoncv]
+
+    def run_after(self):
+        return [SliceLikeToStridedSlice, SsdAnchorMiddleReshape]
+
+    def pattern(self):
+        return dict(
+            nodes=[
+                ('slice_like', dict(op='StridedSlice')),
+                ('slice_like_data', dict(kind='data')),
+                ('reshape0', dict(op='Reshape')),
+                ('reshape0_data', dict(kind='data')),
+                ('concat', dict(op='Concat')),
+            ],
+            edges=[
+                ('slice_like', 'slice_like_data'),
+                ('slice_like_data', 'reshape0'),
+                ('reshape0', 'reshape0_data'),
+                ('reshape0_data', 'concat', {'in': 0}),
+        ])
+
+    def replace_pattern(self, graph: Graph, match: Dict[str, Node]):
+        concat_node = match['concat']
+        concat_name = concat_node.soft_get('name', concat_node.id)
+        if len(concat_node.out_nodes()[0].out_nodes()) == 0:
```

Comment:
rewrite this function using something like `len(concat_node.out_port(0).get_destionations()) == 0`.

BTW, we do we check this?